modules/nixos/boot.nix

```
@@ -10,31 +10,29 @@ in
     default = true;
   };
 
-  config = lib.mkIf cfg.enable (
-    facterLib.canonicalize {
-      boot.initrd.availableKernelModules = builtins.filter (dm: dm != null) (
-        builtins.map
-          (
-            {
-              driver_module ? null,
-              ...
-            }:
-            driver_module
-          )
-          (
-            builtins.filter (
-              with facterLib;
-              isOneOf [
-                # Needed if we want to use the keyboard when things go wrong in the initrd.
-                isUsbController
-                # A disk might be attached.
-                isFirewireController
-                # definitely important
-                isMassStorageController
-              ]
-            ) report.hardware
-          )
-      );
-    }
-  );
+  config = lib.mkIf cfg.enable {
+    boot.initrd.availableKernelModules = builtins.filter (dm: dm != null) (
+      builtins.map
+        (
+          {
+            driver_module ? null,
+            ...
+          }:
+          driver_module
+        )
+        (
+          builtins.filter (
```

```suggestion
          lib.filter (
```

I'd recommend to use `lib` whenever you can, because that way you reduce the risk of your code breaking on older Nix versions. It is also slightly less quirky, because Nixpkgs doesn't have to commit to quirks.

More of a general rule than a problem here, fwiw.

---

Good to know, I'll keep it in mind.